controllers/operator/apigateway_controller.go

```diff
@@ -258,23 +263,20 @@ func apiRulesExist(ctx context.Context, k8sClient client.Client) ([]string, erro
 }
 
 func oryRulesExist(ctx context.Context, k8sClient client.Client) ([]string, error) {
-	// To prevent reconciliation errors during Oathkeeper uninstallation, which arise from the absence of the Oathkeeper CRD,
-	// we exclude calls to the rules from utilizing the cache. This can be achieved by using unstructured objects, as they are excluded from caching.
-	oryRulesList := unstructured.UnstructuredList{}
-	oryRulesList.SetGroupVersionKind(schema.GroupVersionKind{
-		Group:   "oathkeeper.ory.sh",
-		Version: "v1alpha1",
-		Kind:    "rule",
-	})
-
+	oryRulesList := oryv1alpha1.RuleList{}
 	err := k8sClient.List(ctx, &oryRulesList)
+	if meta.IsNoMatchError(err) || apierrors.IsNotFound(err) {
+		// Oathkeeper CRD does not exist, there are no blocking rules
+		return nil, nil
+	}
 	if err != nil {
+		// any other error
 		return nil, err
 	}
 	ctrl.Log.Info(fmt.Sprintf("There are %d ORY Oathkeeper Rule(s) found on cluster", len(oryRulesList.Items)))
 	var blockingOryRules []string
 	for _, rule := range oryRulesList.Items {
-		blocking := rule.GetKind() + "/" + rule.GetName()
+		blocking := rule.GetNamespace() + "/" + rule.GetName()
```

Comment:
I don't know which way is better, but having kind/name format seems to be confusing, as you'd the first to be namespace.
So I agree that this change makes it more obvious, even though we are not sharing what Kind this resource is. We could think about having the resource in format of `Kind Namespace/name`.

For APIRule we still use the kind: https://github.com/kyma-project/api-gateway/blob/b976c7d64c0d45c7020c47e43f78a70ee0a73d6d/controllers/operator/apigateway_controller.go#L251
I think we should have it unified.